Bug 1210305 Lock screen implementation

harman-ankitgupta opened the pull request:
Implemented the lockscreen as per the latest UX. Please review it

harman-ankitgupta commented:
@pgmanutd @Ashish-Developer . Please review it.

TatianaMigacheva commented:
Please change "commit message" to "Bug 1210305 Lock screen implementation". This is Mozilla template.

AlekoNN commented:
@harman-red-square/harman-reviewers

egorlevichev commented:
are changes in 'notifications.js' really required for this bug implementation? From first look they seems more like result of wrong merge/rebase, rather than LS feature implementation

harman-ankitgupta commented:
As per the requirement, similar kind of notification should be clubbed together and notification node is created in the notifications.js. So for combining the notifications, changes in notifications.js should also be done.
Refer UX id: 2.2 for the same.
UX link : http://r1l8an.axshare.com/#p=2_2_open_app_from_lock_screen

egorlevichev commented on apps/system/js/notifications.js:
```diff
@@ -137,7 +137,7 @@ var NotificationScreen = {
             break;
           case 'desktop-notification-close':
-            this.removeNotification(detail.id);
+            this.removeNotification(detail.id, detail.appName);
```
1) It's not clear how it's expected to work, if signature of function 'removeNotification()' still have only 1 parameter?

egorlevichev commented on apps/system/js/notifications.js:
```diff
@@ -380,37 +380,41 @@ var NotificationScreen = {
   addNotification: function ns_addNotification(detail) {
-    // LockScreen window may not opened while this singleton got initialized.
-    this.lockScreenContainer = this.getLockScreenContainer();
+  // LockScreen window may not opened while this singleton got initialized.
```
This 'change' violates Mozilla coding style requirements. And it seems not required for feature implementation, since only formatting was changed. You shall avoid unnecessary changes in your commits, since it makes review of actual changes not trivial.

egorlevichev commented:
please fix inline comments, note that comment regarding changed line indents is common for this patch.

harman-ankitgupta commented:
We have done the method-level line indentation instead of doing it on complete file using two-spaces indent for the same.
If we have to use any other coding style standard,kindly share it with us.

egorlevichev commented:
Since we will contribute all created code to the Mozilla, we shall follow theirs coding style defined here: https://developer.mozilla.org/en-US/Firefox_OS/Developing_Gaia/Understanding_the_Gaia_codebase#Gaia_coding_style which is mostly follows Google's one: http://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml

davidflanagan commented on apps/system/js/notifications.js:
```diff
@@ -672,7 +687,8 @@ var NotificationScreen = {
     var notificationId = notificationNode.dataset.notificationId;
-    this.removeNotification(notificationId);
+    var id = notificationNode.id;
+    this.removeNotification(notificationId, id);
```
It looks like many of the changes in this file are unnecessary changes to the indentation level. This makes it very hard to review. Please fix that before requesting further review.

davidflanagan commented on apps/system/lockscreen/js/lockscreen_inputpad.js:
```diff
@@ -147,7 +147,8 @@
 
   LockScreenInputpad.prototype.updatePassCodeUI =
-  function() {
+  function(keyCode) {
```
I haven't figured out where this keycode is coming from. Where's the code that listens to key events from the keypad?  And are you going to be removing the unnecessary code for displaying and a keypad and handling events on it?

davidflanagan replied:
That doesn't have to be part of this patch, but I need more context information to know what you're doing here so I know what is in scope and what is out of scope for this patch.

davidflanagan commented on apps/system/lockscreen/js/lockscreen_inputpad.js:
```diff
@@ -160,19 +161,32 @@
-      if (span) {
-        if (this.states.passCodeEntered.length > i) {
-          span.dataset.dot = true;
+    var span = this.passcodeCode.childNodes[i];
```
Your indentation is off here, which makes the code very hard to read

davidflanagan commented on apps/system/lockscreen/js/lockscreen_inputpad.js:
```diff
@@ -180,10 +194,11 @@
+            detail: {
+              key: key
+            }
+          }));
```
It appears that there is no change here except to formatting. Your formatting is better, but please don't do that if there are not actually any required changes, because it just makes it harder to review and increases the chances for merge conflicts.

harman-ankitgupta committed:
Bug 1210305 Lock screen implementation

Lock Screen UI implementation as per
http://r1l8an.axshare.com/#p=2_0_lock_screen

Modified Javascript files :-
- notifications.js
- lockscreen_notifications.js
As per the UX, similar kind of notifications should be clubbed together.
Tested by installing an external app and generated local notifications.

- lockscreen_inputpad.js
Password entered by the user on the lockscreen should be visible for the 1
second before changing it to dot.

- lockscreen_state_manager.js
If user clicks on the lockscreen notification and clicks on open after
unlocking the device , the corresponding app should be launched.

AlekoNN commented:
Label is removed since new commit has been added.

Ashish-Developer commented:
label:pr_review_passed

AlekoNN commented:
Let's make the commit towards gaia 2.2r branch and solve the comments there if necessary.
